### ISSUE TITLE
Culling/preview: image can't be completely visible when zooming

### ISSUE THREAD

**Issue by Nilvus** [darktable-org/darktable#6636]
**Describe the bug**

I can't anymore see bottom of the image when zooming on preview or culling view. I've tested that on many images, freshly imported ones or old ones. I don't see that behavior yesterday as I remember, so probably something introduced in recent commits (but I may have missed that before). @AlicVB: any idea? Could you reproduce?

**To Reproduce**

1. Select an image
2. Got preview mode (with Alt+z) or culling one (for example with x)
3. Zoom on image to any zoom level
4. See that you can't reach anymore bottom of the image (part that can be visible depend on where you have point mouse cursor to zoom).

**Expected behavior**
See all image when zooming.

**Platform (please complete the following information):**
 - Darktable Version: 3.3.0+1211~g2fba9d98e
 - OS: Ubuntu 20.10
 - Tested with and without OpenCL: same behavior.

**Comment by ptilopteri** [comment]
also 1211 2fba9d98e but do not see the described problem
openSUSE Tumbleweed
nvidia G04

culling mode I can view fit-screen or x2+ and scroll to top or
bottom ...

**Comment by Nilvus** [comment]
> also 1211 [2fba9d9](https://github.com/darktable-org/darktable/commit/2fba9d98ed6f84f3682f3bb99986bf674844d489) but do not see the described problem
> openSUSE Tumbleweed
> nvidia G04
> 
> culling mode I can view fit-screen or x2+ and scroll to top or
> bottom ...

I don't understand your answer? What the link here with french update?

**Comment by ptilopteri** [comment]
I do not see anything in this issue re: "frency update"

I was commenting about "image can't be completely visible when zooming".

??

**Comment by ptilopteri** [comment]
frency/french

**Comment by Nilvus** [comment]
Forget about that, I find the origin of this issue. It depends on a line in darktablerc: ui/performance=TRUE (if I change it to FALSE, the issue is solved). And that solves also my other issue #6617. So there's something with that setting,

**Comment by Nilvus** [comment]
@jenshannoschwalm: do you have an idea there as, if I'm not wrong, line ui/performance is from your PR #6425? This issue and #6617 are caused by this mode set to true,

**Comment by jenshannoschwalm** [comment]
Default should be false for preferences. Only should be true for low performance systems. There might be scaling missing while presenting the thumb or preview for the issues. Can't test as I have no access to system. Using phone

**Comment by jenshannoschwalm** [comment]
There are several location where ppd had to be corrected. Likely missing still at some places. @AlicVB might help. Unfortunately I'm offline.

**Comment by jenshannoschwalm** [comment]
One more thing I can say here, this is very likely something I missed when doing the hires ppd correction stuff. I can't remember the pr name right now.

**Comment by TurboGit** [comment]
@jenshannoschwalm : yes I have looked at this a bit and I can confirm that this is due to the ppd_thb.

**Comment by TurboGit** [comment]
@jenshannoschwalm : I have added you as assignee to this issue, no luck on my side....

**Comment by jenshannoschwalm** [comment]
Will fix this, (in about 10d)

**Comment by jenshannoschwalm** [comment]
I looked into this and indeed there is a missing correction for ppd_thb in the culling `_event_motion_notify` callback.

In `dtgtk/culling.c` lines ~L.620 we sanitize the positions. After `gtk_widget_get_size_request(th->w_image_box, &iw, &ih);` it should most likely be
```
      const int mindx = iw * darktable.gui->ppd_thb - th->img_width;
      const int mindy = ih * darktable.gui->ppd_thb- th->img_height;
      if(th->zoomx > 0) th->zoomx = 0;
      if(th->zoomx < mindx) th->zoomx = mindx;
      if(th->zoomy > 0) th->zoomy = 0;
      if(th->zoomy < mindy) th->zoomy = mindy;
      l = g_list_next(l);
```

At @TurboGit , as i can't test properly - still in rainy blackforest hiking - could you test please and if this is correct do the pr?

If i understand the code correctly this should fix culling and preview issues ...

**Comment by TurboGit** [comment]
Well how can I say that... I'm very impressed ! I've been contemplating this code trying to fix this issue for some time and did not find the right places for the fix... And you in a middle of a hiking in the in rainy blackforest you found the proper fix. Congrats ! I'll push the fix right now (I've put them as author of this commit, I certainly don't want to get the authorship on this !).

**Comment by Nilvus** [comment]
Good work guys. It's fixed.

**Comment by jenshannoschwalm** [comment]
Lucky punch. I remembered - hiking keeps the mind flowing...
